api/classes/base_controller.js

```diff
@@ -11,133 +13,165 @@ function defaultFormatResponse(model) {
   return model;
 }
 
-function BaseController(model)  {
-  this.Model = model;
-}
+class BaseController {
+  constructor(Model) {
+    this.Model = Model;
+  }
 
-BaseController.prototype.formatRequestData = function(req) {
-  // Abstract base method,
-  // formats data for database entry
-};
+  /* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^request$" }]*/
```

Comment:
eslint complains about the `request` argument in the function signature because it is not used in the function. I keep it in there mainly so that if you decide to override the function (since this function is really an abstract method), you know what param to expect in the function.